io/shared/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -409,6 +412,34 @@ sealed trait Files[F[_]] extends FilesPlatform[F] {
       limit: Long,
       flags: Flags
   ): Pipe[F, Byte, Nothing]
+
+  /** Writes to the specified file as an utf8 string.
+    *
+    * The file is created if it does not exist and is truncated.
+    * Use `writeUtf8(path, Flags.Append)` to append to the end of
+    * the file, or pass other flags to further customize behavior.
+    */
+  def writeUtf8(path: Path): Pipe[F, String, Nothing] = writeUtf8(path, Flags.Write)
+
+  /** Writes to the specified file as an utf8 string using
+    * the specified flags to open the file.
+    */
+  def writeUtf8(path: Path, flags: Flags): Pipe[F, String, Nothing] = in =>
+    in.through(text.utf8.encode).through(writeAll(path, flags))
+
+  /** Writes each string to the specified file as utf8 lines.
+    *
+    * The file is created if it does not exist and is truncated.
+    * Use `writeUtf8Lines(path, Flags.Append)` to append to the end
+    * of the file, or pass other flags to further customize behavior.
+    */
+  def writeUtf8Lines(path: Path): Pipe[F, String, Nothing] = writeUtf8(path, Flags.Write)
+
+  /** Writes each string to the specified file as utf8 lines
+    * using the specified flags to open the file.
+    */
+  def writeUtf8Lines(path: Path, flags: Flags): Pipe[F, String, Nothing] = in =>
+    in.through(text.lines).through(writeUtf8(path, flags))
```

Comment:
As @armanbilge mentioned, if the goal here is to insert a newline between strings, that would be `in => writeUtf8(path, flags)(text.intersperse("\n")`

Reply:
Not sure, does that add a newline at the end? Seems like each string should terminate like that, so that ultimately the file terminates in a newline.

Also, does `text.lines` strip newlines? The reason I ask is because ideally these methods would be inverses of each other.

Reply:
Yep, I'm adding it in a new commit with a couple of unit tests too :)

Reply:
I cried for almost 1 hour in front of failing unit tests and then I saw it:

```scala
def writeUtf8Lines(path: Path): Pipe[F, String, Nothing] = writeUtf8(path, Flags.Write)
                                                                    ^
```

For what is worth, I think that the logic might take in consideration being on windows probably: `text.lines` implementation takes in consideration `\r` too.

> Not sure, does that add a newline at the end? Seems like each string should terminate like that, so that ultimately the file terminates in a newline.

Can we make it pluggable with a parameter maybe?

> Also, does text.lines strip newlines? The reason I ask is because ideally these methods would be inverses of each other.

It does

Reply:
> Also, does text.lines strip newlines? The reason I ask is because ideally these methods would be inverses of each other.

Now that I'm thinking about it, is it the case to add roundtrip property based test like 👇 ?

```scala
forAll { xs: List[String] =>
  Stream.emits(xs).through(Files[IO].writeUtf8Lines(path)) ++
      Files[IO].readUtf8Lines(path).compile.toList.assertEquals(xs)
}
```

Reply:
> For what is worth, I think that the logic might take in consideration being on windows probably: `text.lines` implementation takes in consideration `\r` too.

Now it's handled